pymc_experimental/tests/test_model_builder.py

```diff
@@ -61,6 +65,14 @@ def fitted_model_instance(toy_X, toy_y):
     return model
 
 
+@pytest.fixture
+def fitted_model_instance(fitted_model_instance_base):
+    """Get a fitted model instance. The instance is copied after being fit,
+    so tests using this fixture can modify the model object without affecting
+    other tests."""
+    return copy.deepcopy(fitted_model_instance_base)
```

Comment:
copy doesn't really work for objects that have PyMC models: see https://github.com/pymc-devs/pymc/issues/6985

The approach is not too bad though. What I suggest is to create the idata once and then in this fixture recreate the model and glue-in a copy of the idata. I did something like that with a helper method in this PR: https://github.com/pymc-labs/pymc-marketing/pull/393/commits/44985a82d483d9defe7f80f6600818b64598b6c7

Check the `_build_with_idata` method and how that's used by `thin_fit_result`. Something similar could be used for a `ModelBuilder.copy()`

Reply:
I came up with a workaround for copying the model without using `copy.deepcopy`.

I also noticed that there's a test marked for skipping on win32 due to lack of permissions for temp files, but the marked test doesn't use a temp file. There is a different test that does use a temp file. I thought maybe the annotation got onto the wrong test, so I made a commit to fix that possible issue. If that's wrong or you want to handle it as it's own issue, no problem, I'll take that commit back out.